### PR TITLE
Allow py3.12

### PR DESCRIPTION
**Pull request by felixhekhorn**

### CODE REVIEW

**Comment by felixhekhorn** [comment]
https://github.com/numpy/numpy/issues/24810

File: pyproject.toml [hunk 1]
```diff
@@ -27,8 +27,8 @@ packages = [{ include = "banana", from = "src" }]
 [tool.poetry.dependencies]
 python = "^3.8.0,<3.13"
 ipython = "^8.1.0"
-SQLAlchemy = "^1.4.29,<=2.0"
-numpy = "^1.21.0"
+SQLAlchemy = "^1.4.29"
```
Comment (felixhekhorn) [inline pyproject.toml:30]: That I wouldn't do: https://github.com/N3PDF/banana/issues/22

**Comment by felixhekhorn** [comment]
@alecandido shall we merge?

File: pyproject.toml (hunk 1 above)
Comment (alecandido) [reply pyproject.toml:30]: Why?

The `^...` is already meaning *compatible*, i.e. `>=1.4.29,<2.0`, so it is just redundant.

**Comment by alecandido** [comment]
For me, it's fine:
- Poetry managed to resolve dependencies and lock
- the tests are passing

I'm not sure whether you want to eagerly try some benchmarks, or just wait for them to break (if ever) before attempting a fix.

**Pull request merged**